mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImpl.java

```diff
@@ -199,6 +205,35 @@ public SendOtpResult sendOtp(String relyingPartyId, String clientId, SendOtpDto
         return new SendOtpResult(sendOtpDto.getTransactionId(), maskedEmailId, maskedMobile);
     }
 
+    private boolean knowledgeBaseAuthnetication(KycAuthRequestDto kycAuthRequestDto,IdentityData identityData){
+        String encodedChallenge=kycAuthRequestDto.getKba();
+        try{
+            byte[] decodedBytes = Base64.getUrlDecoder().decode(encodedChallenge);
+            String challenge = new String(decodedBytes, StandardCharsets.UTF_8);
+            Map<String, String> challengeMap = objectMapper.readValue(challenge, Map.class);
+
+            if(challengeMap.containsKey("fullName") && challengeMap.containsKey("dob")){
```

Comment:
Let's not limit the KBA feature to only fullName & dob. KBA form is already configurable in eSignet. The same should be supported in mock implementation.

Reply:
done